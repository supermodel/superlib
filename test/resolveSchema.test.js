const {
  fileSchemaLoader,
  readYAMLFile
} = require('@supermodel/file');
const resolveSchema = require('../src/resolveSchema');

test('resolve valid schema', () => {
  const schema = readYAMLFile('./fixtures/references/AValidB.yaml');
  const loader = fileSchemaLoader(schema['$id'], './fixtures/references/');

  return expect(resolveSchema(schema, loader)).resolves.toMatchSnapshot();
});

test.only('flatten nested definitions into root', () => {
  const schema = readYAMLFile('./fixtures/references/Nested1.yaml');
  const loader = fileSchemaLoader(schema['$id'], './fixtures/references/');

  return expect(resolveSchema(schema, loader)).resolves.toMatchSnapshot();
});