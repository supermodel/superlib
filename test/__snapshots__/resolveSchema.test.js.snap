// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flatten nested definitions into root 1`] = `
Object {
  "$id": "http://supermodel.io/superlib/references/Nested1",
  "definitions": Object {
    "NestedDefinition2": Object {
      "title": "Nested Definition 2",
    },
    "NestedDefinition3": Object {
      "title": "Nested Definition 3",
    },
    "http://supermodel.io/superlib/references/Nested2": Object {
      "$id": "http://supermodel.io/superlib/references/Nested2",
      "properties": Object {
        "nested3": Object {
          "$ref": "Nested3",
        },
      },
      "tilte": "Nested 2",
      "type": "object",
    },
    "http://supermodel.io/superlib/references/Nested3": Object {
      "$id": "http://supermodel.io/superlib/references/Nested3",
      "properties": Object {
        "value": Object {
          "type": "string",
        },
      },
      "tilte": "Nested 3",
      "type": "object",
    },
  },
  "properties": Object {
    "nested2": Object {
      "$ref": "Nested2",
    },
  },
  "tilte": "Nested 1",
  "type": "object",
}
`;

exports[`resolve valid schema 1`] = `
Object {
  "$id": "http://supermodel.io/superlib/references/AValidB",
  "definitions": Object {
    "http://supermodel.io/superlib/references/ValidB": Object {
      "$id": "http://supermodel.io/superlib/references/ValidB",
      "title": "B",
      "type": "object",
    },
  },
  "properties": Object {
    "b": Object {
      "$ref": "ValidB",
    },
  },
  "tilte": "A",
  "type": "object",
}
`;
